### PR TITLE
feat: Convenience BatchDataLoader LoadAsync with multiple keys

### PR DESCRIPTION
**Pull request by benmccallum**
Addresses #1153

### CODE REVIEW

**Comment by benmccallum** [comment]
Toyed with the idea of making it `LoadManyAsync` rather than overloading `LoadAsync`, but just went with what you'd suggested @johnrutherford

**Comment by benmccallum** [comment]
Ready for another review @sungam3r  and @johnrutherford

**Comment by sungam3r** [comment]
Please add also overload with params so one can call `(1, 2, 3)` instead of `new int[] { 1, 2, 3 }`.

**Comment by benmccallum** [comment]
Good idea. Done.

File: src/GraphQL/DataLoader/DataLoaderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        /// <summary>
+        /// Asynchronously load data for the provided given keys
+        /// </summary>
+        /// <param name="dataLoader">The dataloader to use</param>
+        /// <param name="keys">Keys to use for loading data</param>
+        /// <returns>
+        /// A task that will complete when the DataLoader has been dispatched,
+        /// or a completed task if the result is already cached.
+        /// </returns>
+        public static async Task<T[]> LoadAsync<TKey, T>(this IDataLoader<TKey, T> dataLoader, IEnumerable<TKey> keys)
```
Comment (sungam3r) [inline src/GraphQL/DataLoader/DataLoaderExtensions.cs]: Also you can remove async/await from both methods.

File: src/GraphQL/DataLoader/DataLoaderExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        /// </summary>
+        /// <param name="dataLoader">The dataloader to use</param>
+        /// <param name="keys">Keys to use for loading data</param>
+        /// <returns>
+        /// A task that will complete when the DataLoader has been dispatched,
+        /// or a completed task if the result is already cached.
+        /// </returns>
+        public static Task<T[]> LoadAsync<TKey, T>(this IDataLoader<TKey, T> dataLoader, IEnumerable<TKey> keys)
+        {
+            var tasks = new List<Task<T>>(keys.Count());
```
Comment (sungam3r) [inline src/GraphQL/DataLoader/DataLoaderExtensions.cs]: I think that `keys.Count()` call is redundant and can cause the key set to be recalculated if the caller somehow calculates it.  `var tasks = new List<Task<T>>();` will be quite enough.

I'm sure this is the last my remark before merge )

**Comment by sungam3r** [comment]
fixes ##1153

**Comment by benmccallum** [comment]
Resolved.

**Pull request merged**